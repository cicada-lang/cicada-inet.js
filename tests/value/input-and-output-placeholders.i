type Nat -- Type end
node zero -- Nat :value! end
node add1 Nat :prev -- Nat :value! end

node add
  Nat :target!
  Nat :addend
  --------
  Nat :return
end

rule zero add
  (add)-addend
  return-(add)
end

rule add1 add
  (add)-addend inspect (add1)-prev add add1
  (add)-return inspect connect
end

zero add1 zero add1 add inspect
