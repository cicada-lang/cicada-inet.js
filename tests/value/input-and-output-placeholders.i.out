(add₁)-return _placeholder_input_port_for_return_of_add-(_placeholder_input_node_for_return_of_add₁)
(add1₀)-prev _placeholder_output_port_for_prev_of_add1-(_placeholder_output_node_for_prev_of_add1₀)
(_placeholder_output_node_for_addend_of_add₁)-_placeholder_output_port_for_addend_of_add

(add₂)-return prev-(add1₁)
(_placeholder_output_node_for_addend_of_add₁)-_placeholder_output_port_for_addend_of_add addend-(add₂)
(_placeholder_output_node_for_prev_of_add1₀)-_placeholder_output_port_for_prev_of_add1 target-(add₂)
(add1₁)-value
(_placeholder_input_node_for_return_of_add₁)-_placeholder_input_port_for_return_of_add

(add1₃)-value!target-(add₃)
(add1₂)-value addend-(add₃)
(zero₂)-value prev-(add1₃)
(zero₁)-value prev-(add1₂)
(add₃)-return

